assignments/array-methods.js

```diff
@@ -55,29 +55,58 @@ const runners = [{"id":1,"first_name":"Charmain","last_name":"Seiler","email":"c
 
 // ==== Challenge 1: Use .forEach() ====
 // The event director needs both the first and last names of each runner for their running bibs.  Combine both the first and last names into a new array called fullName. 
+
 let fullName = [];
+runners.forEach(function(elem) {
+    fullName.push(`${elem.first_name} ${elem.last_name}`)
+});
 console.log(fullName);
 
 // ==== Challenge 2: Use .map() ====
 // The event director needs to have all the runner's first names converted to uppercase because the director BECAME DRUNK WITH POWER. Convert each first name into all caps and log the result
-let allCaps = [];
+
+let allCaps = runners.map(function (elem) {
+return elem.first_name.toUpperCase();
+})
 console.log(allCaps); 
 
 // ==== Challenge 3: Use .filter() ====
 // The large shirts won't be available for the event due to an ordering issue.  Get a list of runners with large sized shirts so they can choose a different size. Return an array named largeShirts that contains information about the runners that have a shirt size of L and log the result
-let largeShirts = [];
+
+let largeShirts = runners.filter(function(elem) {
+return elem.shirt_size === "L";
+})
 console.log(largeShirts);
 
 // ==== Challenge 4: Use .reduce() ====
 // The donations need to be tallied up and reported for tax purposes. Add up all the donations into a ticketPriceTotal array and log the result
-let ticketPriceTotal = [];
+
+let ticketPriceTotal = runners.reduce(function(total, elem) {
+return total += elem.donation;
```

Comment:
We don't need to do `+=`, only `+`. The `total` variable is discarded after this function is done running, so we don't need to increment it's value. It is the return value that carries across. So we only need to return the sum.